docs/command-line.md

```diff
@@ -418,6 +421,16 @@ processing of output.
 `--use-colour yes` forces coloured output, `--use-colour no` disables coloured
 output. The default behaviour is `--use-colour auto`.
 
+<a id="test-sharding"></a>
+## Test Sharding
+<pre>--shard-count <#number of shards>, --shard-index <#shard index to run></pre>
+When `--shard-count <#number of shards>` is used, the tests to execute will be split evenly in to the given number of sets,
+identified by indicies starting at 0. The tests in the set given by `--shard-index <#shard index to run>` will be executed.
+The default shard count is `1`, and the default index to run is `0`. If the shard index is greater than the number of shards,
+the last shard will be run.
```

Comment:
There is no external reason for this. I made the command line parser report an error if this happens from command line arguments as you suggest, and used an assert in the `createShards` function to ensure this precondition in case the config data is modified directly.